prediction_market_agent/agents/microchain_agent/memory_functions.py

```diff
@@ -18,7 +18,9 @@
 
 class LongTermMemoryBasedFunction(Function):
     def __init__(
-        self, long_term_memory: LongTermMemoryTableHandler, model: str
+        self,
+        long_term_memory: LongTermMemoryTableHandler,
+        model: str = "gpt-4o",  # Use model that works well with these functions, provide other one only if necessary.
```

Comment:
We need `gpt-4o-mini` in these agents because of costs, but mini doesn't create any meaningful summaries/learnings.

So defaulting to `gpt-4o` in these memory functions.